htdocs/user/card.php

```diff
@@ -2791,7 +2792,11 @@
 				// Salary
 				print '<tr><td>'.$langs->trans("Salary").'</td>';
 				print '<td>';
-				print img_picto('', 'salary', 'class="pictofixedwidth paddingright"').'<input size="8" type="text" name="salary" value="'.price2num(GETPOST('salary') ?GETPOST('salary') : $object->salary).'">';
+				if ($caneditsensitivedata) {
```

Comment:
The salary is not a data on the control of the user. So i think it should be managed by the permission on the salary or rh module (generally owned by HR people). Not by a permission on users (generally owned by admin).

Reply:
what do you think about using user->right->salaries->write for the salary field and user->rights->hrm->write_personal_information->write for the others (job, weekly hours, tjm, thm,dateemployment and dateemploymentend) ?

Reply:
@eldy is this Ok for you ?